src/awkward/contents/content.py

```diff
@@ -1296,7 +1296,7 @@ def _is_equal_to(self, other: Self, index_dtype: bool, numpyarray: bool) -> bool
     def _repr(self, indent: str, pre: str, post: str) -> str:
         raise ak._errors.wrap_error(NotImplementedError)
 
-    def _numbers_to_type(self, name: str) -> Self:
+    def _numbers_to_type(self, name: str, including_unknown: bool) -> Self:
```

Comment:
Now the return type changes! 😎 
```suggestion
    def _numbers_to_type(self, name: str, including_unknown: bool) -> Content:
```